### PR TITLE
Use the same code and types for all proof_levels

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR tweaks compilation so no types or important behaviour is determined by the compile-time `proof_level` (except for compile time keys and genesis proof generation). There is still a warning in `Staged_ledger_hash.Non_snark.typ` that is disabled by setting `proof_level < Full`, but this should have no other impact.

This should (hopefully!) make it possible to switch the integration test executive to a lower proof level and avoid downloading keys.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

**Review by nholland94 — APPROVED** [review]
:+1: if this works, that'll be amazing for int test framework iteration speed.

File: src/lib/ledger_proof/dune [hunk 1]
```diff
@@ -2,6 +2,5 @@
  (name ledger_proof)
  (public_name ledger_proof)
  (libraries core_kernel mina_base transaction_snark)
- (preprocessor_deps "../../config.mlh")
  (instrumentation (backend bisect_ppx))
- (preprocess (pps ppx_optcomp ppx_bin_prot ppx_sexp_conv ppx_hash ppx_compare ppx_version ppx_deriving_yojson)))
+ (preprocess (pps ppx_bin_prot ppx_sexp_conv ppx_hash ppx_compare ppx_version ppx_deriving_yojson)))
```
Comment (psteckler) [inline src/lib/ledger_proof/dune:6]: can these be combined into `ppx_jane`, now that `ppx_optcomp` is removed?

not a big deal, either way, tho

File: src/lib/ledger_proof/ledger_proof.ml [hunk 2]
```diff
@@ -69,19 +66,8 @@ struct
   let create ~statement ~sok_digest ~proof:_ = (statement, sok_digest)
 end
 
-[%%if
-proof_level = "full"]
-
 include Prod
```
Comment (psteckler) [inline src/lib/ledger_proof/ledger_proof.ml:69]: is the `Debug` module dead code? still keep it for debugging?

**Review by psteckler — APPROVED** [review]

File: src/lib/ledger_proof/ledger_proof.ml (hunk 2 above)
Comment (mrmr1993) [reply src/lib/ledger_proof/ledger_proof.ml:69]: I suppose it is, yea. I'll do that in a follow-up

**Comment by aneesharaines** [comment]
!approved-for-mainnet

**Pull request merged**